### ISSUE TITLE
[PostgreSQL] Precision issue with float values

### ISSUE THREAD

**Issue by rbovard** [dbeaver/dbeaver#5376]
This query did not return the same result for float fields:

```
SELECT id, hauteur_moyenne, hauteur_mediane, hauteur_min, hauteur_max
FROM my_table;
```

pgAdmin (OK):

```
id  |hauteur_moyenne|hauteur_mediane|hauteur_min|hauteur_max
----|---------------|---------------|-----------|-----------
1847|           11.4|           11.5|          0|       13.1
```

DBeaver (NOK):

```
id  |hauteur_moyenne|hauteur_mediane|hauteur_min|hauteur_max
----|---------------|---------------|-----------|-----------
1847|     11.3999996|           11.5|          0| 13.1000004
```

Table definition:

```
CREATE TABLE my_table (
	id serial NOT NULL,
	hauteur_moyenne float4 NULL,
	hauteur_mediane float4 NULL,
	hauteur_min float4 NULL,
	hauteur_max float4 NULL,
	CONSTRAINT my_table_pkey PRIMARY KEY (id)
);

INSERT INTO my_table(hauteur_moyenne, hauteur_mediane, hauteur_min, hauteur_max)
VALUES (11.4, 11.5, 0, 13.1);
```

**Comment by p91paul** [comment]
pgadmin is most likely rounding your value when presenting it, since 0.1 and 0.4 are periodic in binary. 7 decimal numbers should be quite close to the float precision, it makes sense.

**Comment by serge-rider** [comment]
Indeed, PostgreSQL driver sets float4 scale to 8 digits. DBeaver tries to do its best.
Workaround: disable usage of database provided scale + reduce number of max fraction digits:

![image](https://user-images.githubusercontent.com/6398845/53693753-b2d21c80-3db5-11e9-8327-7c5e4d361ff2.png)

**Comment by rbovard** [comment]
Okay, thanks to both of you for your explanations and workaround.

You can close this issue as your wish.

**Comment by serge-rider** [comment]
General issue with float4 formatting will be fixed in #6111
